editoast/src/views/train_schedule/mod.rs

```diff
@@ -628,26 +629,30 @@ async fn create_backend_request_payload(
     scenario: &Scenario,
     db_pool: Data<DbPool>,
 ) -> Result<SimulationRequest> {
+    let mut db_conn = db_pool.get().await?;
+
     let infra = scenario.infra_id.expect("Scenario should have an infra");
     let rolling_stocks_ids = train_schedules
         .iter()
         .map(|ts| ts.rolling_stock_id)
         .collect::<HashSet<_>>();
-    let mut rolling_stocks = Vec::new();
+    let mut rolling_stocks: Vec<RollingStock> = Vec::new();
     let mut rolling_id_to_name = HashMap::new();
-    for rolling_stock_id in rolling_stocks_ids {
-        let rolling_stock = RollingStockModel::retrieve(db_pool.clone(), rolling_stock_id)
-            .await?
-            .ok_or(TrainScheduleError::RollingStockNotFound { rolling_stock_id })?;
-        rolling_id_to_name.insert(
-            rolling_stock_id,
-            rolling_stock
-                .name
-                .clone()
-                .expect("Rolling stock shoud have a name"),
-        );
+
+    let rolling_stock_batch: Vec<RollingStockModel> =
+        RollingStockModel::retrieve_batch_or_fail(&mut db_conn, rolling_stocks_ids, |missing| {
+            let first_missing_id = missing.iter().next().unwrap_or(&0);
```

Comment:
```suggestion
            let first_missing_id = missing.iter().next().expect("<some msg>");
```

If missing is empty it's a ModelV2 bug and we want to catch the bug.